src/tests/unit/common/visualization-scan-result-store-data-builder.ts

```diff
@@ -24,6 +24,7 @@ export class VisualizationScanResultStoreDataBuilder extends BaseDataBuilder<IVi
     }
 
     public withSelectorMap(type: VisualizationType, selectorMap: any): VisualizationScanResultStoreDataBuilder {
+        // tslint:disable-next-line: switch-default
```

Comment:
>// tslint:disable-next-line: switch-default [](start = 8, length = 43)

I think it might be better to just make tests fail in the unsupported cases, rather than suppressing the rule, eg:

```
            default:
                throw new Error(`Test logic error: VisualizationType ${type} does not support fullAxeResultsMap`);
```

Reply:
We are passing `-1 as VisualizationType` in so many places that this is not an option for me at this time! 

I'm considering other options as @Shobhit1 also raise the flag on this.

Reply:
I think I'm maybe misunderstanding; I just tried out using this strategy and there were no instances where a test failed due to passing -1. However, there *were* a few instances of failures that look like they're exposing legitimate test bugs (that is, a test's logic is incorrectly assuming it should be able to use this builder on a visualization type that isn't supported), eg, chrome-command-handler.test.ts at line 130 has an each case that assumes VisualizationStoreDataBuilder.withDisable support TabStops even though it doesn't.

Even if there was such a case, I think it would still be a legitimate test bug we should look into fixing; if a test is invoking this builder and we're silently throwing away the input, it probably means the test isn't testing what it thinks it is.

Can you show a specific example where using a `throw`ing default case in the instances this PR is touching causes a failure for a test using the -1 thing in a reasonable way?

---
In reply to: [267848311](https://github.com/Microsoft/accessibility-insights-web/pull/404#discussion_r267848311) [](ancestors = 267848311)

Reply:
@dbjorge  I'm with you in the fact that having an explicit default -> throw statement will help us catch issues. So that's what I did.